cms/static/js/spec/views/course_video_settings_spec.js

```diff
@@ -281,8 +302,286 @@ define(
+                expect($courseVideoSettingsEl.find('.organization-credentials-wrapper:visible')).not.toExist();
+            });
+
+            it('shows organization credentials when organization credentials for selected provider are not present', function() {
+                renderCourseVideoSettingsView(null, transcriptionPlans);
+
+                // Check Cielo24 provider
+                changeProvider('Cielo24');
+
+                // Verify organization credentials are shown.
+                expect($courseVideoSettingsEl.find('.organization-credentials-wrapper:visible')).toExist();
+
+                // Verify transcript preferences are not shown.
+                expect($courseVideoSettingsEl.find('.course-video-transcript-preferances-wrapper:visible')).not.toExist();
+            });
+
+            it('shows transcript preferences when organization credentials for selected provider are present', function() {
+                renderCourseVideoSettingsView(null, transcriptionPlans, transcriptOrganizationCredentials);
+
+                // Check Cielo24 provider
+                changeProvider('Cielo24');
+
+                // Verify organization credentials are not shown.
+                expect($courseVideoSettingsEl.find('.organization-credentials-wrapper:visible')).not.toExist();
+
+                // Verify transcript preferences are shown.
+                expect($courseVideoSettingsEl.find('.course-video-transcript-preferances-wrapper:visible')).toExist();
+            });
+
+            it('shows organization credentials view if clicked on change provider button', function() {
+                // Verify organization credentials view is not shown initially.
+                expect($courseVideoSettingsEl.find('.organization-credentials-content')).not.toExist();
+
+                // Click change button to render organization credentials view.
+                $courseVideoSettingsEl.find('.action-change-provider').click();
+
+                // Verify organization credentials is now shown.
+                expect($courseVideoSettingsEl.find('.organization-credentials-content')).toExist();
+            });
+
+            it('shows api secret input field if selected provider is 3Play Media', function() {
+                // Set selected provider to 3Play Media
+                courseVideoSettingsView.selectedProvider = '3PlayMedia';
+
+                // Click change button to render organization credentials view.
+                $courseVideoSettingsEl.find('.action-change-provider').click();
+
+                // Verify 3play api secret is present.
+                expect(
+                    $courseVideoSettingsEl.find('.' + courseVideoSettingsView.selectedProvider + '-api-secret')
+                ).toExist();
+
+                // Also verify api key is present.
+                expect(
+                    $courseVideoSettingsEl.find('.' + courseVideoSettingsView.selectedProvider + '-api-key')
+                ).toExist();
```

Comment:
Above `expect`s should go into a util and then we should use that util everywhere we expect org credentials to be rendered.